### PR TITLE
Feature: Display nametags next to player heads when in a group

### PR DESCRIPTION
**Pull request by ly-schneider**
This is a simple feature to improve QOL. An example use case would be when on a big server where not all the player heads are known and instead of trying to search on the tab list the names would be displayed next to it. 

As you can see I also implemented toggles to disable this feature or only display the nametags when talking.

Thank you for this great mod and I hope you like my QOL improvement!

Bildschirmfoto 2024-10-28 um 20 18 32
Bildschirmfoto 2024-10-28 um 20 12 31
Bildschirmfoto 2024-10-28 um 20 12 04
Bildschirmfoto 2024-10-28 um 20 11 37

https://github.com/user-attachments/assets/019f1a61-fe7a-45d2-8d48-119851d9b343

### CODE REVIEW

File: common/src/main/java/de/maxhenkel/voicechat/gui/VoiceChatSettingsScreen.java [hunk 1]
```diff
@@ -89,13 +89,15 @@ protected void onUpdate(AudioType type) {
...
             addRenderableWidget(Button.builder(ADJUST_VOLUMES, button -> {
                 minecraft.setScreen(new AdjustVolumesScreen());
-            }).bounds(guiLeft + 10, y, xSize - 20, 20).build());
-            y += 21;
+            }).bounds(guiLeft + xSize / 2 + 1, y, (xSize - 20) / 2 - 1, 20).build());
         }
+        y += 21;
+        addRenderableWidget(new ToggleDisplayNametagsButton(guiLeft + 10, y, (xSize - 20) / 2 - 1, 20));
+        addRenderableWidget(new ToggleOnTalkNametagsButton(guiLeft + xSize / 2 + 1, y, (xSize - 20) / 2 - 1, 20));
+        y += 21;
```
Comment (henkelmax) [inline common/src/main/java/de/maxhenkel/voicechat/gui/VoiceChatSettingsScreen.java:100]: This setting should not be in this menu. The cloth config menu would make much more sense.

File: common/src/main/java/de/maxhenkel/voicechat/voice/client/KeyEvents.java [hunk 2]
```diff
@@ -55,9 +57,11 @@ public static void registerKeyBinds() {
...
         KEY_ADJUST_VOLUMES = ClientCompatibilityManager.INSTANCE.registerKeyBinding(new KeyMapping("key.voice_chat_adjust_volumes", InputConstants.UNKNOWN.getValue(), "key.categories.voicechat"));
+        KEY_DISPLAY_NAMETAGS = ClientCompatibilityManager.INSTANCE.registerKeyBinding(new KeyMapping("key.toggle_nametags", InputConstants.UNKNOWN.getValue(), "key.categories.voicechat"));
+        KEY_ON_TALK_NAMETAGS = ClientCompatibilityManager.INSTANCE.registerKeyBinding(new KeyMapping("key.toggle_on_talk_nametags", InputConstants.UNKNOWN.getValue(), "key.categories.voicechat"));
 
         ALL_KEYS = new KeyMapping[]{
-                KEY_PTT, KEY_WHISPER, KEY_MUTE, KEY_DISABLE, KEY_HIDE_ICONS, KEY_VOICE_CHAT, KEY_VOICE_CHAT_SETTINGS, KEY_GROUP, KEY_TOGGLE_RECORDING, KEY_ADJUST_VOLUMES
+                KEY_PTT, KEY_WHISPER, KEY_MUTE, KEY_DISABLE, KEY_HIDE_ICONS, KEY_VOICE_CHAT, KEY_VOICE_CHAT_SETTINGS, KEY_GROUP, KEY_TOGGLE_RECORDING, KEY_ADJUST_VOLUMES, KEY_DISPLAY_NAMETAGS, KEY_ON_TALK_NAMETAGS
```
Comment (henkelmax) [inline common/src/main/java/de/maxhenkel/voicechat/voice/client/KeyEvents.java:64]: There is no reason to have this as a keybind. People won't change this setting that often.

File: common/src/main/resources/assets/voicechat/lang/de_de.json [hunk 3]
```diff
@@ -189,5 +197,7 @@
   "cloth_config.voicechat.config.group_hud_icon_scale": "Gruppenoberflächen Symbol Skalierung",
   "cloth_config.voicechat.config.group_player_icon_pos_x": "Gruppenoberflächen Symbol X position",
   "cloth_config.voicechat.config.group_player_icon_pos_y": "Gruppenoberflächen Symbol Y position",
-  "cloth_config.voicechat.config.show_own_group_icon": "Eigenes Gruppenoberflächen Symbol anzeigen"
+  "cloth_config.voicechat.config.show_own_group_icon": "Eigenes Gruppenoberflächen Symbol anzeigen",
+  "cloth_config.voicechat.config.display_nametags": "Namensschilder anzeigen",
+  "cloth_config.voicechat.config.on_talk_nametags": "Namensschilder während Gespräche"
```
Comment (henkelmax) [inline common/src/main/resources/assets/voicechat/lang/de_de.json:202]: The translations seem to be made with a translator. Please just don't do any translations for languages you don't know.

**Comment by Mike74846** [comment]
I love this idear for all the versions and I find for server with friends is this good. And the GUI buttons was so good. Easy and for the Voice Chat Setting the perfect area. I hope he add this for all versions

File: common/src/main/resources/assets/voicechat/lang/en_us.json [hunk 4]
```diff
@@ -208,5 +216,9 @@
   "cloth_config.voicechat.config.group_player_icon_pos_y": "Group chat HUD icon Y position",
   "cloth_config.voicechat.config.group_player_icon_pos_y.description": "The Y position of the player icons in the group chat HUD.\nNegative values mean anchoring to the bottom instead.",
   "cloth_config.voicechat.config.show_own_group_icon": "Show yourself in the group chat HUD",
-  "cloth_config.voicechat.config.show_own_group_icon.description": "If your own player icon should be displayed in the group chat HUD when you are in a group."
+  "cloth_config.voicechat.config.show_own_group_icon.description": "If your own player icon should be displayed in the group chat HUD when you are in a group.",
```
Comment (henkelmax) [inline common/src/main/resources/assets/voicechat/lang/en_us.json:219]: You didn't add these options to the cloth config menu

File: common/src/main/java/de/maxhenkel/voicechat/voice/client/GroupChatManager.java [hunk 5]
```diff
@@ -116,4 +128,10 @@ public static List<PlayerState> getGroupMembers(boolean includeSelf) {
         return entries;
     }
 
+    private static void displayNametag(GuiGraphics guiGraphics, Minecraft mc, PlayerState state, int namePosX) {
+        guiGraphics.pose().pushPose();
+        guiGraphics.pose().scale(0.5F, 0.5F, 1F);
+        guiGraphics.drawString(mc.font, state.getName(), namePosX, 7, FastColor.ARGB32.color(255, 255, 255, 255), false);
```
Comment (henkelmax) [inline common/src/main/java/de/maxhenkel/voicechat/voice/client/GroupChatManager.java:134]: The name being completely white isn't really good for readability.

**Review by henkelmax — CHANGES_REQUESTED** [review]
Please target your PR at the main branch. If you want to implement such a feature, you should also do PRs to all Minecraft versions that the mod is currently released and actively developed for. (If its not easily cherry-pickable)

**Pull request closed without merging**